### PR TITLE
GnuTests fixes and updates

### PR DESCRIPTION
**Pull request by rivy**
* removes the `t*` factor tests (leaving the mutator code as a comment for future use if/when needed)
* merges GnuTests job steps into single job which tests and produces coverage (should reduce some CI pressure)
* misc minor updates

Note: this will *fail* with a decreased PASS count of -18 for the now missing 18 of 20 passing `t*` factor tests, but should recalibrate from that point after merged into 'main'.

### CODE REVIEW

**Comment by sylvestre** [comment]
In the future, can you try do smaller PR ?
I was expecting one to just disable factor.

> change/CICD ~ (GnuTests) merge coverage/non-coverage jobs

For example, I would like to avoid this change. I do see the value for them to be separated.
The one without coverage takes an hour. I can use that to make a decision to accept/reject a PR. The coverage takes also 3 hours...

The rest seems great, well done :)

**Review by tertsdiepraam — COMMENTED** [review]
Agreed with @sylvestre on not merging coverage and the rest, but the rest looks good!

**Comment by rivy** [comment]
The PR has a small line count change and I believe it generally functions as a single logical piece, just separated into component pieces so that I can modify the PR more easily.

It is a bit more "wide" because I started with an experimental commit set from a couple of weeks prior.

I'd say that only the first commit, increasing the visible size of the SHA IDs, is semi-unrelated. That's easy to remove to another PR. And I'm happy to split this up more if you'd like...

> change/CICD ~ (GnuTests) merge coverage/non-coverage jobs

I can easily add back a non-coverage job, but I'm unsure of it's utility.

After disabling the extra factor tests, from the runs that I have on my fork and the run here, it looks like the coverage step takes about 65 min and the non-coverage version takes about 55 min; so, now only about a 10 min difference. And it drops that 55 min job step from using up CI resources.

Given the minimal difference and the amount of duplicated code that I was syncing between steps, I didn't think the non-coverage code was worth the continued maintenance effort. Does the non-coverage code give some info that I'm not seeing?

I am happy to add it back in if you both would like it back; I just don't understand the need.

**Comment by rivy** [comment]
I pared a couple of the marginally-related commits out.
I post them as a separate CICD refactor/fixup.

**Comment by sylvestre** [comment]
> I am happy to add it back in if you both would like it back; I just don't understand the need.

Because currently, we have currently 52m vs 2h50m ;)

image

On the last main run:
https://github.com/uutils/coreutils/actions/runs/2067897268
Maybe because of factor, we see an improvement ? (this is also why I don't like PR with unrelated changes: it is harder to pin point improvement and regressions)

**Comment by rivy** [comment]
So that you can compare it, I'll put it back in as an add-on commit (b/c I think you'll want it taken back out again).

[Fork run with both](https://github.com/rivy/rs.coreutils/actions/runs/2048166520)

* Run GNU tests (55m 6s)
* Run GNU tests with coverage (1h 9m)

[Run for this PR]<https://github.com/uutils/coreutils/actions/runs/2066126283>

* Run GNU tests (with coverage) (1h 5m)

**Comment by sylvestre** [comment]
ok, as I am spending a lot of time waiting for GNU's test results (to the point that I have been considering renting a machine to make this faster), I would prefer to keep them separated. Sorry

File: util/build-gnu.sh [hunk 1]
```diff
@@ -70,17 +70,37 @@ sed -i 's|^"\$@|/usr/bin/timeout 600 "\$@|' build-aux/test-driver
...
-    for i in $(seq -w 0 20); do
-        make "tests/factor/t${i}.sh"
-    done
-    sed -i -e 's|sha1sum |/usr/bin/sha1sum |' tests/factor/t*sh
-fi
+# Handle generated factor tests
+t_first=00
+t_max=36
+# t_max_release=20
+# if test "${UU_MAKE_PROFILE}" != "debug"; then
```
Comment (sylvestre) [inline util/build-gnu.sh:77]:
```suggestion
# Disable factor tests, too many intermittent issues and they
# are not providing much values
# if test "${UU_MAKE_PROFILE}" != "debug"; then
```

**Comment by sylvestre** [comment]
Great side effect, the log size dropped from 3,2G => 2.4M :)

**Pull request closed without merging**